SS14.Launcher/ViewModels/HubSettingsViewModel.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using DynamicData;
+using Splat;
+using SS14.Launcher.Models.Data;
+using SS14.Launcher.Utility;
+
+namespace SS14.Launcher.ViewModels;
+
+public class HubSettingsViewModel : ViewModelBase
+{
+    public ObservableCollection<HubViewModel> HubList { get; set; } = new();
+
+    private readonly DataManager _dataManager = Locator.Current.GetRequiredService<DataManager>();
+
+    public void Save()
+    {
+        var hubs = new List<Hub>();
+
+        for (var i = 0; i < HubList.Count; i++)
+        {
+            var uri = new Uri(HubList[i].Address, UriKind.Absolute);
+
+            // Automatically add trailing slashes for the user
+            if (!uri.AbsoluteUri.EndsWith("/"))
+            {
+                uri = new Uri(uri.AbsoluteUri + "/", UriKind.Absolute);
```

Comment:
Wouldn't it make more sense to do this check *before* running it through `new Uri()`?

Reply:
I figured this was more robust, specifically since it would trim whitespace but maybe there's other edge cases I haven't thought of. And since it's only done when the user presses OK in the menu, I imagine it doesn't matter for performance in practice?

Reply:
Meh it's minor so I guess do whatever.